onnxconverter_common/float16.py

```diff
@@ -121,6 +121,42 @@ def convert_float_to_float16(model, min_positive_val=1e-7, max_finite_val=1e4):
     if func_infer_shape is not None:
         model = func_infer_shape(model)
     queue.append(model)
+    name_mapping = {}
+    graph_io_to_skip = set()
+    io_casts = set()
+    if keep_io_types:
+        for i, n in enumerate(model.graph.input):
+            if n.type.tensor_type.elem_type == onnx_proto.TensorProto.FLOAT:
+                output_name = 'graph_input_cast_' + str(i)
+                name_mapping[n.name] = output_name
+                graph_io_to_skip.add(n.name)
+
+                node_name = 'graph_input_cast' + str(i)
+                new_value_info = model.graph.value_info.add()
+                new_value_info.CopyFrom(n)
+                new_value_info.name = output_name
+                new_value_info.type.tensor_type.elem_type = onnx_proto.TensorProto.FLOAT16
+                # add Cast node (from tensor(float) to tensor(float16) after graph input
+                new_node = [helper.make_node('Cast', [n.name], [output_name], to=10, name=node_name)]
+                model.graph.node.extend(new_node)
+                io_casts.add(node_name)
+
+        for i, n in enumerate(model.graph.output):
```

Comment:
nit: refactor it as a function?

Reply:
Agreed.  I'll do that with the next change.